### PR TITLE
Implement logging with optional hslogger dependency

### PR DESCRIPTION
**Pull request by fendor**

### CODE REVIEW

File: hie-bios.cabal [hunk 1]
```diff
@@ -18,6 +18,10 @@ Extra-Source-Files:     ChangeLog
 
 tested-with: GHC==8.6.4, GHC==8.4.4, GHC==8.2.2
 
+flag hslogger
```
Comment (fendor) [inline hie-bios.cabal]: maybe bad name for a flag?

**Comment by ndmitchell** [comment]
Why make hslogger optional? It's not a huge dependency, and certainly making it mandatory isn't going to upset clients like ghcide.

**Comment by fendor** [comment]
@ndmitchell if the additional dependencies are alright, including network and network-bsd, we can add it as a required dependency. opinion, @mpickering ?

**Comment by ndmitchell** [comment]
We already have `hslogger` as a dependency due to `haskell-lsp`.

**Pull request merged**